submissions/UA-girl/task_friends_app/js/app.js

```diff
@@ -0,0 +1,354 @@
+        const [backPage, forwPage] = document.querySelectorAll('.button-page');
+        if ((this.currentPage + 1) * CARDS_PER_PAGE >= this.cards.length)  {
+            forwPage.classList.add('disabled');
+            forwPage.setAttribute('disabled', true);
+        } else {
+            forwPage.removeAttribute('disabled');
+            forwPage.classList.remove('disabled');
+        }
+        if (this.currentPage === 0) {
+            backPage.classList.add('disabled');
+            backPage.setAttribute('disabled', true);
+        } else {
+            backPage.classList.remove('disabled');
+            backPage.removeAttribute('disabled');
+        }
+        const span = document.querySelector('.page');
+        span.textContent = this.currentPage + 1;
+    }
+
+    resetApp() {
+        this.cards = cards.slice();
+        this.resetFilterState(true);
+        this.resetSortingState();
+        document.querySelector('.label-range').innerHTML = `Max age: 100`;
+        document.querySelector('.range').value = 100;
+    }
+
+    displayPeople() {
+        this.container.innerHTML = '';
+        let cardsRender;
+        if (this.cards.length > CARDS_PER_PAGE) {
+            cardsRender =
+                this.cards.slice(CARDS_PER_PAGE * this.currentPage, CARDS_PER_PAGE * (this.currentPage + 1));
+        } else {
+            cardsRender = this.cards.slice();
+        }
+        const fragment = document.createDocumentFragment();
+        cardsRender.forEach(person => fragment.append(person.card));
+        this.container.appendChild(fragment);
+        this.disableEnablePageToggler();
+    }
+
+    addListenersFilters() {
+        const app = this;
+        filterSet.addEventListener('input', function (event) {
+            const target = event.target.closest('input');
+            if (!target) {
+                return
+            }
+            const name = target.getAttribute('name');
+            const value = target.value;
+            switch (name) {
+                case 'search':
+                    app.nameFilter(value);
+                    break;
+                case 'sorting':
+                    app.sortCards(value, target.dataset.name);
+                    break;
+                case 'gender':
+                    const wasToggled = app.filterStates.gender.toggled;
+                    app.genderFilter(value, wasToggled);
+                    break;
+                case 'range':
+                    app.ageFilter(value);
+                    break;
+            }
+            app.currentPage = 0;
+            app.displayPeople();
+        });
+
+        filterSet.addEventListener('click', function ({target}) {
+            target = target.closest('input.button-reset');
+            if (!target) {
+                return
+            }
+            app.currentPage = 0;
+            app.resetApp();
+            app.displayPeople();
+        })
+    }
+
+    addEventListenersPages() {
```

Comment:
Consider more informative function name here, please. This function name should be clear. 
Now it's not clear enough to follow what this function really do.